### PR TITLE
refactor: use generated integration constants

### PR DESCRIPTION
**Pull request by koladilip**
## PR Description

I have added the generated destination constants file and updated the integration constants to use it. Additionally, I have replaced the hard-coded destination names and display names with these constants using a script that is also in PR.

Removed ComScore which is an unused integration.

## Linear task (optional)
Resolves INT-2930
https://linear.app/rudderstack/issue/INT-2930/add-github-actions-to-generate-files-and-raise-prs

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Sanity Suite

- [x] All sanity suite test cases pass locally

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

## Release Notes

- **New Features**
	- Introduced a new `Destinations.ts` file containing constants for various analytics and marketing integrations.
	- Added a script to automate updates of integration name constants across multiple files.

- **Refactor**
	- Updated multiple integration constants files to import `NAME` and `DISPLAY_NAME` from the new `Destinations.ts` instead of local declarations.

- **Tests**
	- Added a new test suite to validate integration constants against the `Destinations` file, ensuring consistency.

### CODE REVIEW

**Review by saikumarrs — CHANGES_REQUESTED** [review]

File: packages/analytics-js-common/src/constants/Destinations.ts
Comment (saikumarrs) [reply packages/analytics-js-common/src/constants/Destinations.ts:4]: I guess this will be fixed later once we make it automatic. So, feel free to resolve it for now.

Comment (koladilip) [reply packages/analytics-js-common/src/constants/Destinations.ts:4]: it is intentional that we want to depend on development as during development we need it, we don't need to wait till it is released to develop JS device mode integration.

File: packages/analytics-js-common/src/constants/Destinations.ts
Comment (koladilip) [reply packages/analytics-js-common/src/constants/Destinations.ts:1]: Comscore device mode is there but enabled so my script skipped it

Comment (koladilip) [reply packages/analytics-js-common/src/constants/Destinations.ts:1]: Sure

**Review by saikumarrs — CHANGES_REQUESTED** [review]
Let's revert the changes in the `examples` directory.

File: packages/analytics-js-common/src/constants/Destinations.ts
Comment (saikumarrs) [reply packages/analytics-js-common/src/constants/Destinations.ts:4]: Do we want to merge the integrations config PR before the integration is ready? The changes will be automatically released in its next cycle. If the integration PR is lagging behind, it could create problems.

Moroever, further such changes to the destination names in a given release cycle, could lead to multiple PRs in this repo. If not attended to properly, will lead to a mess.

I'd rather suggest to regenerate this file with every production release. The new destinations that are part of the release can be configured (I guess through the options object) to be hidden by default on our dashboard.

Once the integration is released in the SDK, we can create a hotfix to only unhide the destination in the dashboard.

Let's discuss this offline if needed.

Comment (koladilip) [reply packages/analytics-js-common/src/constants/Destinations.ts:4]: We need to ensure there is only one PR at any given time, either by removing duplicates or updating the same PR. The idea is that once changes are merged into the `develop` branch, they can be immediately utilized in the SDK for further development. There’s no need to wait for the release process, as having the code in the SDK doesn’t cause any issues since we still control it through the Integration configuration. Merging into `develop` allows development to start right away without waiting for the release to complete.

CC: @krishna2020 your thought?

**Review by saikumarrs — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**
